src/v/pandaproxy/kafka_client_cache.h

```diff
@@ -60,6 +61,14 @@ class kafka_client_cache {
     size_t size() const;
     size_t max_size() const;
 
+    template<std::invocable<kafka_client_cache&> Func>
+    auto invoke_on_cache(Func&& func) {
+        return ss::try_with_gate(_gc_gate, [this, &func] {
```

Comment:
@graphcareful is right about the lifetime here of `func`. you need to move it through the captures all the way down into the capture that eventually invokes it.